### ISSUE TITLE
Update cursor to indicate pending language server work

### ISSUE THREAD

**Issue by maxdeviant** [zed-industries/zed#13545]
### Check for existing issues

- [X] Completed

### Describe the feature

Filing this on behalf of @fasterthanlime ([link](https://twitter.com/fasterthanlime/status/1805936516792803546)):

<img width="601" alt="Screenshot 2024-06-26 at 10 17 09 AM" src="https://github.com/zed-industries/zed/assets/1486634/4fae01bd-8b45-405c-bcf4-2e6eb8947041">

Textual version of the tweet, for searchability:

> I suddenly find myself wishing that the cursor (text cursor, the block) went spinny / indicated loading state in some way when there's a pending LSP request (like "finding all references") — it feels too quiet right now, I have to check the status bar to make sure

**Comment by jamiebuilds** [comment]
It would be nice if this was implemented with a slight delay before it changed the cursor so it didn't flash it for a just a couple frames when something runs quickly. If an LSP request takes more than say 200ms (which is hardly even a perceptible delay even though a flash of something is perceptible) you could then flip it to a loading state

**Comment by GioPan04** [comment]
I was trying to implement this, the Linux part was fairly easy, the problem is with the macOS part.
Turns out that you can't show the progress cursor via the [`NSCursor`](https://developer.apple.com/documentation/appkit/nscursor) class. macOS will show the progress cursor only if the app froze. So I don't know if it make sense to implement this only on Linux (and maybe Windows) because it will create a feature disparity

**Comment by fasterthanlime** [comment]
To be clear, I really meant the text cursor, not the mouse cursor!

(since the text cursor is usually where my gaze actually is)

**Comment by jackTabsCode** [comment]
Yeah, I don't want my mouse cursor to change. I think animating the buffer cursor could be cool!
